overlord/snapstate/component.go

```diff
@@ -78,33 +80,58 @@ func InstallComponentPath(st *state.State, csi *snap.ComponentSideInfo, info *sn
 		},
 	}
 
-	return doInstallComponent(st, &snapst, compSetup, snapsup, "")
+	componentTS, err := doInstallComponent(st, &snapst, compSetup, snapsup, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return componentTS.taskSet(), nil
 }
 
 type componentInstallFlags struct {
-	RemoveComponentPath bool `json:"remove-component-path,omitempty"`
-	SkipProfiles        bool `json:"skip-profiles,omitempty"`
+	RemoveComponentPath    bool `json:"remove-component-path,omitempty"`
+	SkipProfiles           bool `json:"skip-profiles,omitempty"`
+	SkipKernelModulesSetup bool `json:"skip-kernel-modules,omitempty"`
```

Comment:
Maybe these two could be merged in one as they should be set in the same cases. Or maybe they are not needed at all as you can determine if the snap revision is changed or not, as if it changed we do not want to run these tasks. There is the other case of skipping when we want to run these tasks only once when installing multiple components but the snap does not change, but I think that in that case it would be cleaner to pass as argument the task to inject in the middle instead of creating it.

Reply:
Thanks, but isn't `JointSnapComponentsInstall` the same as `!snapst.IsInstalled() || snapst.Current != snapSi.Revision`, that is, this is a snap install or a snap refresh? Can it be simplified further?

Reply:
No, not quite unfortunately, since we allow `snap refresh --revision=n snapname`, where `n` is the current revision of the snap. There is a similar case for installing a snap from file, where the snap is the same revision as the snap that is installed.

Reply:
Oh, right, I forgot about this case :slightly_frowning_face: